cmd/lakectl/cmd/store/factory.go

```diff
@@ -0,0 +1,61 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+)
+
+var (
+	ErrNoStorageAdapter = errors.New("no storage adapter found")
+)
+
+type ObjectStoreEntry struct {
+	FullKey     string
+	RelativeKey string
+	Address     string
```

Comment:
Please add some documentation with examples on what you expect these to contain.